tests/test_events/handler/tabEventHandler.test.js

```diff
@@ -168,4 +168,103 @@ describe("addTab", () => {
         expect(tabData.tabs[1].file_id).toBeNull();
         expect(tabData.tabs[1].path).toBe(filePath2[1]);
     });
+
+     // Should discard invalid tab and send nothing
+     test("Discard unrecognized tab and send nothing", () => {
+        /*
+        - Should discard invalid tab
+        - Should not create .yml file
+        - Should not send any data
+        */
+        configUtil.addRepo();
+        addUser(baseRepoPath, true);
+        const mockTabs = [
+            {
+                tabs: [
+                    {
+                        input: undefined,
+                        isActive: false,
+                    },
+                ]
+            }
+        ]
+        Object.defineProperty(vscode.window.tabGroups, 'all', {
+            get: jest.fn(() => mockTabs),
+        });
+        const handler = new tabEventHandler(repoPath);
+        // Record timestamp
+        const createdAt = formatDatetime(new Date().getTime());
+        handler.handleTabChangeEvent(createdAt)
+        let tabFiles = fs.readdirSync(tabsRepo)
+        // Assert file should not be created
+        expect(tabFiles).toHaveLength(0);
+    });
+
+     // Should discard invalid tab and send valid tabs
+     test("Discard unrecognized tab and send valid tabs", () => {
+        /*
+        - Should discard invalid tab
+        - Should create .yml file
+        - Should send any data for valid tabs
+        */
+        const tabCreationTime = new Date();
+        configUtil.addRepo();
+        addUser(baseRepoPath, true);
+        const config_data = readYML(configPath);
+        const tab1_fileId = config_data.repos[repoPath].branches[DEFAULT_BRANCH]["file_1.js"]
+        const repo_id = config_data.repos[repoPath].id;
+        const mockTabs = [
+            {
+                tabs: [
+                    {
+                        input: {
+                            uri: {
+                                path: newFilePath1,
+                            }
+                        },
+                        isActive: true,
+                    },
+                    {
+                        input: {
+                            uri: {
+                                path: newFilePath2,
+                            }
+                        },
+                        isActive: false,
+                    },
+                    {
+                        input: undefined,
```

Comment:
I think as per the error, it should be 

```
{
                        input: {
                            uri: undefined
                        },
                        isActive: true,
                    },
```
Thoughts? @saadberry

Reply:
Actually we were getting the value of `input` as `undefined`, so current implementation is fine.

But logically, your solution can work as well as we will discard this tab.

Reply:
ref:

Updating body as per your suggestion:
![image](https://github.com/user-attachments/assets/b5e0e51d-d146-43bf-98de-0de861407826)

Screenshot of file tests run:
![image](https://github.com/user-attachments/assets/56ddf63e-2331-4395-9889-58cfe4fb7ea1)

Reply:
If "input" is coming as undefined, error should be "Cannot read property `uri` of undefined", right? But the actual error says different @saadberry

Reply:
@saadberry As in this SS, input is NOT undefined, however it does not have a property "uri", which makes 
`input.uri` undefined and hence `input.url.path` breaks. Makes sense?

![image](https://github.com/user-attachments/assets/d2be0ab8-2998-40e0-8bbf-3bced0152917)

Reply:
Kindly correct me if I am referring something different here

Reply:
@Basit-qc `input` itself seems to be undefined

e.g. for the following tabs:

![image](https://github.com/user-attachments/assets/e546c475-e35e-466f-9d20-2558a732e44e)

Getting undefined `input`:
![image](https://github.com/user-attachments/assets/daf8b0e9-0682-4b57-8c38-c0e041b6a43e)


> I think as per the error, it should be

Reply:
Hmm, could it be that since `input` is mentioned (even though it is `undefined`). Code was trying to find `input.uri`?
> If "input" is coming as undefined, error should be "Cannot read property uri of undefined", right? But the actual error says different @saadberry

Reply:
Updated checks in `extension.ts` and `tab_event_handler` as:

- `if  (!changeEvent.changed[0]?.input || !changeEvent.changed[0]?.input?.uri || !changeEvent.changed[0]?.input?.uri.path) return`
- `if (!tab.input || !tab.input?.uri || !tab.input?.uri.path) return null;`

Updated assertions as:
1. Invalid tab, undefined input value - should not send anything
2. Invalid tab, undefined uri value- should not send anything
3. Invalid tab, undefined path value - should not send anything

Test results:

![image](https://github.com/user-attachments/assets/3330be8a-7194-44ff-9e15-45ea9ac96d77)

Reply:
But the error says "reading `path`" , not "reading `uri`"

> Hmm, could it be that since `input` is mentioned (even though it is `undefined`). Code was trying to find `input.uri`?
> 
> > If "input" is coming as undefined, error should be "Cannot read property uri of undefined", right? But the actual error says different @saadberry

Reply:
You are right. I am not a 100% sure about this

Reply:
But I believe our checks should prevent that error from occuring:

>Updated checks in extension.ts and tab_event_handler as:
>if (!changeEvent.changed[0]?.input || !changeEvent.changed[0]?.input?.uri || >!changeEvent.changed[0]?.input?.uri.path) return
>if (!tab.input || !tab.input?.uri || !tab.input?.uri.path) return null;